Fantasy-Quest/Assets/Scripts/Dialogue/DialogueSpeaker.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using Sirenix.OdinInspector;
+using Subtitles;
+using UnityEngine;
+
+namespace Dialogue
+{
+    [RequireComponent(typeof(Collider2D))]
+    [RequireComponent(typeof(AudioSource))]
+    [AddComponentMenu("Scripts/Dialogue/Dialogue.DialogueSpeaker")]
+    public class DialogueSpeaker : MonoBehaviour, ISpeakable
+    {
+        [Header("Speech")]
+        [SerializeField]
+        private List<Replica> firstTrySpeech;
+
+        [SerializeField]
+        private List<Replica> alternativeSpeech;
+
+        [Space]
+        [Header("Components")]
+        [SerializeField]
+        [ValidateInput(nameof(HasISubtitlesView), "GameObject must have ISubtitlesView")]
+        private GameObject subtitlesViewGameObject;
+
+        private Coroutine sayCoroutine;
+        private ISubtitlesView subtitlesView;
+        private Voice voice;
+        private bool wasSaid;
+
+        private void Awake()
+        {
+            if (!subtitlesViewGameObject.TryGetComponent(out subtitlesView))
+            {
+                Debug.Log($"{subtitlesView} is NULL\n{GetType()} callback in {gameObject.name}");
```

Comment:
Log error here and use [error context](https://docs.unity3d.com/ScriptReference/Debug.LogError.html)